### DISCUSSION TITLE
Does Hooks exist?

### DISCUSSION THREAD

**Discussion by hmmhmmhm** [nanojsx/nano#10] [Help / Questions] [answered]
Is there a smaller hook or life cycle is exist?
I know this is excessive greed, but I'm still curious. 🤔

**Comment by hm-lee** [comment]
oh i founded `useState` are exist!, so is there can be makes useEffect...?

**Reply by yandeu** [reply]
No, there is no `useEffect` hook. Lifecycles are only possible in `class components`.

**Comment by yandeu** [comment, marked as answer]
You can use hooks. The only hook provided is `useState`, but you can of course build and use your own hooks.

It is not possible to use lifecycles on functional components.

The way I build the library, `class components` manage state and have lifecycles; `functional components` just render thinks. (Just like in the old days of react)
